tools/unitctl/unitctl/src/wait.rs

```diff
@@ -8,113 +8,91 @@ use unit_client_rs::unitd_instance::UnitdInstance;
 /// Waits for a socket to become available. Availability is tested by attempting to access the
 /// status endpoint via the control socket. When socket is available, ControlSocket instance
 /// is returned.
-pub async fn wait_for_socket(cli: &UnitCtl) -> Result<ControlSocket, UnitctlError> {
-    // Don't wait, if wait_time is not specified
-    if cli.wait_time_seconds.is_none() {
-        return cli.control_socket_address.instance_value_if_none().await.and_validate();
+pub async fn wait_for_sockets(cli: &UnitCtl) -> Result<Vec<ControlSocket>, UnitctlError> {
+    let socks: Vec<ControlSocket>;
+    match &cli.control_socket_addresses {
+        None => {
+            socks = vec![find_socket_address_from_instance().await?];
+        },
+        Some(s) => socks = s.clone(),
     }
 
-    let wait_time =
-        Duration::from_secs(cli.wait_time_seconds.expect("wait_time_option default was not applied") as u64);
-    let max_tries = cli.wait_max_tries.expect("max_tries_option default was not applied");
-
-    let mut attempt: u8 = 0;
-    let mut control_socket: ControlSocket;
-    while attempt < max_tries {
-        if attempt > 0 {
-            eprintln!(
-                "Waiting for {}s control socket to be available try {}/{}...",
-                wait_time.as_secs(),
-                attempt + 1,
-                max_tries
-            );
-            std::thread::sleep(wait_time);
+    let mut mapped = vec![];
+    for addr in socks {
+        if cli.wait_time_seconds.is_none() {
+            mapped.push(addr.to_owned().validate()?);
+            continue;
         }
 
-        attempt += 1;
-
-        let result = cli.control_socket_address.instance_value_if_none().await.and_validate();
+        let wait_time =
+            Duration::from_secs(cli.wait_time_seconds.expect("wait_time_option default was not applied") as u64);
+        let max_tries = cli.wait_max_tries.expect("max_tries_option default was not applied");
```

Comment:
is it customary to use `expect` in cli applications? Panicking here is not as big of a deal as in a service application. If the cli panics, are there any drawbacks to that?

Reply:
No, It basically just exits the flow and prints to stderr